test: Eliminate writing to /usr

martinpitt opened the pull request:
This is generally error prone, and specifically bad on OSTree. Use bind
mounts instead to fake things.

Make check-packages @nondestructive, so that restore_dir() will do the
temporary bind mount for us. That test is also super-quick (boot time
dominated the test time).

Make netlib.py helpers nondestructive as well, which will help with
making bond and vlan tests fully nondestructive.

mvollmer commented on test/verify/netlib.py:
```diff
@@ -136,16 +136,18 @@ def iface_con_id(self, iface):
-        m.write("/etc/NetworkManager/conf.d/99-dhcp.conf", "[main]\ndhcp=dhclient\n")
-        m.execute("systemctl restart NetworkManager")
+        self.write_file("/etc/NetworkManager/conf.d/99-dhcp.conf", "[main]\ndhcp=dhclient\n")
+        self.machine.execute("systemctl try-restart NetworkManager")
```
We need to restart NM after restoring the file for non-destructive tests, right?

martinpitt replied:
That's right. This is done in PR #13879 in a more general way. So in that sense this change would be superfluous, but also harmless.

martinpitt commented:
The [slow checkpoint rollbacks](https://logs.cockpit-project.org/logs/pull-13877-20200409-103021-1725705c-fedora-31/log.html#130-2) fail a little too often now, I'll have a look.

martinpitt commented:
Ah, SELinux was throwing a spanner into the works:

    kernel: audit: type=1400 audit(1586435605.290:386): avc:  denied  { execute } for  pid=4169 comm="NetworkManager" name="dhclient" dev="dm-0" ino=10908988 scontext=system_u:system_r:NetworkManager_t:s0 tcontext=unconfined_u:object_r:var_lib_t:s0 tclass=file permissive=0

Fixed. @mvollmer, I also reverted the ensure_nm_uses_dhclient() change for now, as you mentioned it was incomplete and we don't need it after all. PTAL?
